tests/integrationv2/bin/SSLSocketClient.java

```diff
@@ -37,8 +37,11 @@ public static void main(String[] args) throws Exception {
             socket.startHandshake();
             System.out.println("Starting handshake");
 
-            while (stdIn.read(buffer) != -1) {
-                out.write(buffer);
+            while (true) {
+                int read = stdIn.read(buffer);
+                if (read == -1)
+                    break;
```

Comment:
Why use while(true) instead of while(stdIn.read)?

Reply:
There is a possibility that read returns 0 bytes. In that case we used to send the previous contents of the buffer. This was causing us to send N bytes and receive N + 100 bytes (100 being the size of the buffer).

This change lets us send exactly the bytes read.

Reply:
Oh I see now! You just switched to while(true) because you needed to save the "read" value for use in the write. You could do `while((read = stdIn.read(buffer)) != -1)`, but I don't think that's any clearer.